zk.js/src/account.ts

```diff
@@ -418,14 +419,19 @@ export class Account {
     }
   }
 
-  private addPrivateKey(proofInput: any, inputUtxos: Utxo[]) {
-    proofInput["inPrivateKey"] = inputUtxos.map((utxo: Utxo) => {
-      if (utxo.publicKey === this.keypair.publicKey.toString()) {
-        return this.keypair.privateKey;
-      } else {
-        return STANDARD_COMPRESSION_PRIVATE_KEY;
-      }
-    });
+  private addPrivateKey(
+    proofInput: any,
+    inputUtxos: (Utxo | ProgramUtxo<PlaceHolderTData>)[],
+  ) {
+    proofInput["inPrivateKey"] = inputUtxos.map(
+      (utxo: Utxo | ProgramUtxo<PlaceHolderTData>) => {
+        if ("data" in utxo) {
```

Comment:
Why change this?
We are using the standard key pair also for filling utxos which do not have data.

Reply:
Actually I think a better way is two ifs (not sure whether the code below works)
```suggestion
        if (STANDARD_COMPRESSION_PUBLIC_KEY === utxo.owner) { 
```

Reply:
else should throw an error